lightning/src/ln/channel.rs

```diff
@@ -4669,7 +4675,11 @@ impl<Signer: Sign> Writeable for Channel<Signer> {
 
 		self.channel_update_status.write(writer)?;
 
-		write_tlv_fields!(writer, {(0, self.announcement_sigs, option)});
+		write_tlv_fields!(writer, {
+			(0, self.announcement_sigs, option),
+			(1, self.minimum_depth, option),
+			(3, self.counterparty_selected_channel_reserve_satoshis, option),
```

Comment:
Just so I'm clear, we are writing these twice (here and above), right? Wouldn't we want to increment our version instead?

Reply:
By writing it twice (with an odd TLV) old versions can always read the `Channel`, and will do so correctly. New versions will let the TLV value override the old copy, allowing for a distinction between `Some(0)` and `None`, while old versions will use 0 for both.

I'm not really sure what the criteria should be for bumping the version number - my gut was kinda "only if we need to to understand the new fields correctly, which I think you can argue is fine here - the old code will run fine no matter what value is serialized, though if we start using `Some(0)` to mean something, then we need to bump the version so that we can flag to old versions that they don't know what our minimum_depth means.

I added a check at open to reject minimum_depth of 0 and a comment noting that the serialization version should be updated.

Reply:
> By writing it twice (with an odd TLV) old versions can always read the Channel, and will do so correctly. New versions will let the TLV value override the old copy, allowing for a distinction between Some(0) and None, while old versions will use 0 for both.

Could be useful to document this in a comment

>if we start using Some(0) to mean something, then we need to bump the version so that we can flag to old versions that they don't know what our minimum_depth means.


And this